homeassistant/components/camera/__init__.py

```diff
@@ -614,6 +678,36 @@ def async_update_token(self) -> None:
             hashlib.sha256(_RND.getrandbits(256).to_bytes(32, "little")).hexdigest()
         )
 
+    async def async_internal_added_to_hass(self) -> None:
+        """Run when entity about to be added to hass."""
+        await super().async_internal_added_to_hass()
+        # Note: State is always updated by entity on return
+        await self.async_refresh_providers()
+
+    async def async_refresh_providers(self) -> bool:
+        """Determine if any of the registered providers are suitable for this entity.
+
+        This affects state attributes, so it should be invoked any time the registered
+        providers or inputs to the state attributes change.
+
+        Returns True if any state was updated (and needs to be written)
+        """
+        old_state = self._rtsp_to_webrtc
+        self._rtsp_to_webrtc = await self._async_use_rtsp_to_webrtc()
+        return old_state != self._rtsp_to_webrtc
```

Comment:
Why doesn't this update state if needed?

Reply:
This is called from two contexts, one when being added to hass and one when adding the registration. The thought here was that when being added to hass it already updates the state in the context this is called with the comment above (`# Note: State is always updated by entity on return`).

One option is to always update here, though my general impression was that folks want to be self conscious of state updates -- maybe its not as bad as I realize. Or another option is to do this another way, if you have a suggestion.

Reply:
I think we should keep the state update control inside the entity as it knows when that's needed.

If want to avoid an extra state update we could add a parameter to not do the state update. The double update would only happen when adding the entity so also not crucial to avoid.

Reply:
Thanks, addressing in https://github.com/home-assistant/core/pull/63183